src/zap/zapinfo.cpp

```diff
@@ -3307,6 +3307,36 @@ unsigned ZapInfo::getClassSize(CORINFO_CLASS_HANDLE cls)
     return size;
 }
 
+unsigned ZapInfo::getHeapClassSize(CORINFO_CLASS_HANDLE cls)
+{
+#ifdef FEATURE_READYTORUN_COMPILER
+    if (IsReadyToRunCompilation())
+    {
+        _ASSERTE(!"getHeapClassSize");
+        ThrowHR(E_NOTIMPL);
+    }
+#endif
+
+    DWORD size = m_pEEJitInfo->getHeapClassSize(cls);
+
+    return size;
+}
+
+BOOL ZapInfo::classHasFinalizer(CORINFO_CLASS_HANDLE cls)
+{
+#ifdef FEATURE_READYTORUN_COMPILER
+    if (IsReadyToRunCompilation())
```

Comment:
It would be better to name this method `canAllocateOnStack` instead to encapsulate the policy on the EE-size.

The classes from the same version bubble can be allocated on stack just fine, even with R2R.

Reply:
It's not enough that the class being stack-allocated is from the same version bubble. I believe we need to check IsInheritanceChainLayoutFixedInCurrentVersionBubble here.

Reply:
It is fine to just return false for R2R here, for now.

Reply:
Do you think returning true if IsInheritanceChainLayoutFixedInCurrentVersionBubble returns true is incorrect?

Reply:
> Do you think returning true if IsInheritanceChainLayoutFixedInCurrentVersionBubble returns true is incorrect?

Yes, it should be correct. Feel free to add it if you would like to make work properly right away.

Reply:
Done.